Sources/Environment/Environment.swift

```diff
@@ -44,17 +44,18 @@ public struct Environment {
     public func all() -> [String: String] {
         var env = environ
         var pairs: [String: String] = [:]
-        while true {
-            guard let cval = env?.pointee else { break }
-            let pairString = String(cString: cval)
+        while let cpair = env?.pointee {
+            defer { env = env?.successor() }
+
+            let pairString = String(cString: cpair)
             let pair = pairString
-                .characters
+                .unicodeScalars
```

Comment:
My (very limited) understanding of unicode scalars is that it separates i.e. `=́` into `=` + `\u{301}`, which means that in some cases, splitting behavior could be undefined. 

```swift
let str = "thing1\u{003D}\u{301}thing2"
let split = str.unicodeScalars.split(separator: "=")
split[0]// "thing1"
split[1]// "́thing2" (notice the tick on the first quote)
```

This is the only case where it should matter (`=` with any sort of accent), but that might be a deal breaker (which is fine, I don't insist).

The reason I use unicode scalars is because the overhead of converting to and from `characters` is avoided, giving a very significant (about 3x) performance boost. Though, to be fair, the only other time I have used them is in [Zewo's TrieRouteMatcher](https://github.com/Zewo/TrieRouteMatcher/blob/master/Source/TrieRouteMatcher.swift#L126), where splitting on characters was _the_ bottleneck (about 30% of cpu time).

Some code to compare performance:
```swift
import Foundation

let str = "\u{1F600}=\u{1F4A9}" // 😀=💩

// unicode scalars
let then1 = NSDate()
for _ in 0..<100000 {
    let split = str.unicodeScalars.split(separator: "=")
    split[0]// "thing1"
    split[1]// "thing2"
}
let now1 = NSDate()
let delta1 = now1.timeIntervalSince1970 - then1.timeIntervalSince1970
print("unicodeScalars: \(delta1)")

// characters
let then2 = NSDate()
for _ in 0..<100000 {
    let split = str.characters.split(separator: "=")
    split[0]// "thing1"
    split[1]// "thing2"
}
let now2 = NSDate()
let delta2 = now2.timeIntervalSince1970 - then2.timeIntervalSince1970
print("characters: \(delta2)")
```

Compiled with:
```sh
xcrun -sdk macosx swiftc main.swift -O -o main
```

Outputs:
```
unicodeScalars: 0.11971116065979
characters: 0.37518310546875
```

But, I don't insist. If you feel like the performance does not outweigh the safety, your call.